cpp/ps0004/ps0004_solution.cpp

```diff
@@ -2,39 +2,32 @@
 using namespace std;
 
 int main() 
-
 {
-    int input;
-
-
+    // prompting user for number 1/1.5
+    float input;
     cout << "Enter a number: ";
+    cin >> input;
 
-
-    if (!(cin >> input)) 
+    // checking invalidaty of user's input 3/4
+    if (input == 0) 
     {
         cout << "Invalid character/s or zero value!" << endl;
-            }
+    }
 
-    if (input >= 32 && input <= 126) 
+    else if (static_cast<int>(input) != input) 
     {
-        cout << "The character for " << input << " is " << static_cast<char>(input) << "." << endl;
-
+        cout << "Invalid: a floating point number!" << endl;
     }
-
+    
     else if (input < 32 || input > 126) 
-
     {
         cout << "Invalid: number is out of the range!" << endl;
     }
-
-    else if (static_cast<int>(input) != input) 
-    {
-        cout << "Invalid: a floating point number!" << endl;
-    }
-
-    else 
+    
+    else
     {
-        cout << "Invalid character/s or zero value!" << endl;
+        // printing the number and the opposite printable character 2/2
+        cout << "The character for " << input << " is " << static_cast<char>(input) << "." << endl;
     }
-
-}
\ No newline at end of file
+}
+// good work 👍 6.5/10

```

Comment:
- [x] program is in the correct path
- [ ] no syntax errors, bugs or warnings while compilation and execution
- [ ] formatted code to be clear and readable
- [ ] commented some phrases for blocks of code
- [ ] main returns non-zero/zero number according to the case

you can check the [problem solution](https://onlinegdb.com/9x6wLTlkK) for more